### PR TITLE
RR-676 - Populated "Withdrawn Courses" tab

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR populates the Withdrawn courses tab

### Prisoner with Withdrawn courses
Screenshot 2024-03-25 at 16 36 04

### Prisoner with no Withdrawn courses
Screenshot 2024-03-25 at 17 05 09

### CODE REVIEW

File: server/@types/viewModels/index.d.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ declare module 'viewModels' {
     coursePlannedEndDate?: Date
     isAccredited: boolean
     grade?: string
+    withdrawalReason?: string
```
Comment (nathanrussell-moj-digital) [inline server/@types/viewModels/index.d.ts:118]: I had to add withdrawal reason to the view model, as it wasn't there previously; hence the changes to the mapper and it's tests below

File: server/filters/formatCuriousCourseStatusFilter.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export default function formatCuriousCourseStatusFilter(value: string): string {
+  const curiousCourseStatusValue = CuriousCourseStatusValues[value as keyof typeof CuriousCourseStatusValues]
+  return curiousCourseStatusValue
+}
+
+enum CuriousCourseStatusValues {
```
Comment (nathanrussell-moj-digital) [inline server/filters/formatCuriousCourseStatusFilter.ts:6]: New filter for the Curious course status values. It is used on the Withdrawn course tab for differentiating between Withdrawn and Temporarily Withdrawn; but we could use the filter on the other tabs if the design changes

File: server/views/pages/inPrisonCoursesAndQualifications/_completedCoursesTable.njk [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<table class="govuk-table" data-module="moj-sortable-table" data-qa="completed-courses-sortable-table">
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/inPrisonCoursesAndQualifications/_completedCoursesTable.njk:1]: I've refactored the macro (now deleted 👇 ) into 3 simple nunjucks includes files.
It felt simpler this way, especially as there are subtle differences between each tab (completed, in progress, and withdrawn) when rendering the same field (start date; and initial sort order)

**Review by chrimesdev — APPROVED** [review]
LGTM

**Pull request merged**